plugins/defaultplugins/l3plugin/arp_config.go

```diff
@@ -0,0 +1,189 @@
+// are compared with the VPP run-time configuration and differences are applied through the VPP binary API.
+type ArpConfigurator struct {
+	Log logging.Logger
+
+	GoVppmux    govppmux.API
+	ArpIndexes  idxvpp.NameToIdxRW
+	ArpIndexSeq uint32
+	SwIfIndexes ifaceidx.SwIfIndex
+	vppChan     *govppapi.Channel
+
+	Stopwatch *measure.Stopwatch
+}
+
+// Init initializes ARP configurator
+func (plugin *ArpConfigurator) Init() (err error) {
+	plugin.Log.Debug("Initializing ArpConfigurator")
+
+	// Init VPP API channel
+	plugin.vppChan, err = plugin.GoVppmux.NewAPIChannel()
+	if err != nil {
+		return err
+	}
+
+	err = plugin.checkMsgCompatibility()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TransformArp converts raw entry data to ARP object
+func TransformArp(arpInput *l3.ArpTable_ArpTableEntry, index ifaceidx.SwIfIndex, log logging.Logger) (*vppcalls.ArpEntry, error) {
+	if arpInput == nil {
+		log.Infof("ARP input is empty")
+		return nil, nil
+	}
+	if arpInput.Interface == "" {
+		log.Infof("ARP input does not contain interface")
+		return nil, nil
+	}
+	if arpInput.IpAddress == "" {
+		log.Infof("ARP input does not contain IP")
+		return nil, nil
+	}
+	if arpInput.PhysAddress == "" {
+		log.Infof("ARP input does not contain MAC")
+		return nil, nil
+	}
+
+	ifName := arpInput.Interface
+	ifIndex, _, exists := index.LookupIdx(ifName)
+	if !exists {
+		return nil, fmt.Errorf("ARP entry interface %v not found", ifName)
+	}
+
+	ipAddr := net.ParseIP(arpInput.IpAddress)
+	macAddr, err := net.ParseMAC(arpInput.PhysAddress)
+	if err != nil {
+		return nil, err
+	}
+	arp := &vppcalls.ArpEntry{
+		Interface:  ifIndex,
+		IPAddress:  ipAddr,
+		MacAddress: macAddr,
+		Static:     arpInput.Static,
+	}
+	return arp, nil
+}
+
+// AddArp processes the NB config and propagates it to bin api call
+func (plugin *ArpConfigurator) AddArp(entry *l3.ArpTable_ArpTableEntry) error {
+	//plugin.Log.Infof("Creating new ARP entry %v -> %v (%v) for interface %v", entry.IpAddress, entry.PhysAddress, entry.Static, entry.Interface)
+	plugin.Log.Infof("Creating ARP entry %+v", *entry)
+
+	// Transform route data
+	arp, err := TransformArp(entry, plugin.SwIfIndexes, plugin.Log)
+	if err != nil {
+		return err
+	}
+	if arp == nil {
+		return nil
+	}
+	plugin.Log.Debugf("adding ARP: %+v", *arp)
+
+	// Create and register new route
+	err = vppcalls.VppAddArp(arp, plugin.vppChan, measure.GetTimeLog(ip.IPNeighborAddDel{}, plugin.Stopwatch))
+	if err != nil {
+		return err
+	}
+	arpID := arpIdentifier(arp.Interface, arp.IPAddress.String(), arp.MacAddress.String())
+	plugin.ArpIndexes.RegisterName(arpID, plugin.ArpIndexSeq, nil)
+	plugin.ArpIndexSeq++
+	plugin.Log.Infof("ARP entry %v registered", arpID)
+
+	return nil
+}
+
+// ChangeArp processes the NB config and propagates it to bin api call
+func (plugin *ArpConfigurator) ChangeArp(entry *l3.ArpTable_ArpTableEntry, prevEntry *l3.ArpTable_ArpTableEntry) error {
+	return fmt.Errorf("CHANGE ARP NOT IMPLEMENTED")
```

Comment:
why? is it not supported on VPP? If that is the case, we can do the delete + add here.